starboard/nplb/player_write_sample_test.cc

```diff
@@ -102,16 +103,176 @@ TEST_P(SbPlayerWriteSampleTest, WriteMultipleBatches) {
+    return;
+  }
+
+  const SbTime kDurationToPlay = kSbTimeSecond;
+  const float kSegmentSize = 0.1f;
+
+  GroupedSamples samples;
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamples(
+        0, player_fixture.ConvertDurationToVideoBufferCount(kDurationToPlay));
+    samples.AddVideoEOS();
+  }
+
+  int total_buffers_to_write =
+      player_fixture.ConvertDurationToAudioBufferCount(kDurationToPlay);
+  for (int i = 0; i < total_buffers_to_write; i++) {
+    SbTime current_timestamp = player_fixture.GetAudioSampleTimestamp(i);
+    SbTime next_timestamp = player_fixture.GetAudioSampleTimestamp(i + 1);
+    SbTime buffer_duration = next_timestamp - current_timestamp;
+    SbTime segment_duration = buffer_duration * kSegmentSize;
+    SbTime written_duration = 0;
+    while (written_duration < buffer_duration) {
+      samples.AddAudioSamples(
+          i, 1, written_duration, written_duration,
+          std::max<int64_t>(
+              0, buffer_duration - written_duration - segment_duration));
+      written_duration += segment_duration;
+    }
+  }
+  samples.AddAudioEOS();
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerPresenting());
+
+  SbTime start_system_time = SbTimeGetMonotonicNow();
+  SbTime start_media_time = player_fixture.GetCurrentMediaTime();
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
+
+  SbTime end_system_time = SbTimeGetMonotonicNow();
+  SbTime end_media_time = player_fixture.GetCurrentMediaTime();
+
+  const SbTime kDurationDifferenceAllowance = kSbTimeSecond;
+  ASSERT_NEAR(end_media_time, kDurationToPlay, kDurationDifferenceAllowance);
+  ASSERT_NEAR(end_system_time - start_system_time + start_media_time,
+              kDurationToPlay, kDurationDifferenceAllowance);
+}
+
+TEST_P(SbPlayerWriteSampleTest, PartialAudioDiscardAll) {
+  if (!IsPartialAudioSupported()) {
+    // TODO: Use GTEST_SKIP when we have a newer version of gtest.
+    SB_LOG(INFO)
+        << "The platform doesn't support partial audio. Skip the tests.";
+    return;
+  }
+
+  SbPlayerTestFixture player_fixture(GetParam());
+  if (HasFatalFailure()) {
+    return;
+  }
+  if (!player_fixture.HasAudio()) {
+    // TODO: Use GTEST_SKIP when we have a newer version of gtest.
+    SB_LOG(INFO) << "Skip PartialAudio test for audioless content.";
+    return;
+  }
+
+  const SbTime kDurationToPlay = kSbTimeSecond;
+  const SbTime kDurationPerWrite = 100 * kSbTimeMillisecond;
+  const SbTime kNumberOfBuffersToDiscard = 100;
+
+  GroupedSamples samples;
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamples(
+        0, player_fixture.ConvertDurationToVideoBufferCount(kDurationToPlay));
+    samples.AddVideoEOS();
+  }
+
+  int written_buffer_index = 0;
+  SbTime written_duration = 0;
+  int num_of_buffers_per_write =
+      player_fixture.ConvertDurationToAudioBufferCount(kDurationPerWrite);
```

Comment:
As this is partial audio test, we should some how convert the |num_of_buffers_per_write| back to duration and store it in a variable named |duration_per_write|, so we can adjust it to be the same as the duration of the first |num_of_buffers_per_write| access units.

This will also make lines like line 249 more accurate.